lib/protocol/message.go

```diff
@@ -129,9 +129,17 @@ type ClusterConfigMessage struct {
 	ClientName    string   // max:64
 	ClientVersion string   // max:64
 	Folders       []Folder // max:1000000
+	Flags         uint32
 	Options       []Option // max:64
 }
 
+type DownloadProgressMessage struct {
+	Folder  string                       // max:64
+	Updates []FileDownloadProgressUpdate // max:1000000
```

Comment:
We should decide whether this is all about DownloadProgress or FileDownloadProgress. I think DownloadProgress suffices.